unity/Assets/Scripts/PhysicsRemoteFPSAgentController.cs

```diff
@@ -4638,20 +4687,35 @@ public void PlaceHeldObject(string objectId, bool forceAction=false, bool placeS
             InstantiatePrefabTest script = physicsSceneManager.GetComponent<InstantiatePrefabTest>();
             //set degreeIncrement to 90 for placing held objects to check for vertical angles
             List<ReceptacleSpawnPoint> spawnPoints = targetReceptacle.ReturnMySpawnPoints(onlyPointsCloseToAgent);
-            if (randomSeed != 0) {
-                List<ReceptacleSpawnPoint> randomizedSpawnPoints = new List<ReceptacleSpawnPoint>();
-                float maxDistance = z;
-                if (maxDistance == 0.0f) {
+            if (randomSeed != 0 || putNearXY) {
+                List<KeyValuePair<ReceptacleSpawnPoint, float>> distSpawnPoints = new List<KeyValuePair<ReceptacleSpawnPoint, float>>();
+                if (Mathf.Approximately(maxDistance, 0.0f)) {
                     maxDistance = maxVisibleDistance;
                 }
                 foreach (ReceptacleSpawnPoint sp in spawnPoints) {
                     Vector3 tmp = new Vector3(transform.position.x, sp.Point.y, transform.position.z);
                     if (Vector3.Distance(sp.Point, tmp) < maxDistance) {
-                        randomizedSpawnPoints.Add(sp);
+                        float dist = 0; 
+                        if (putNearXY) {
+                            dist = -Vector3.Distance(sp.Point, hit.point);  // Will be used to sort from greatest negative distance (closest) to least (farthest) from Raycast Hit
```

Comment:
This seems backwards to me? Sorting, by default, goes from smallest to largest, this will make it go from largest to smallest.

Reply:
this has been resolved in comment below.